server/source/source.controller.ts

```diff
@@ -103,6 +103,23 @@ export class SourceController {
         );
     }
 
+    @ApiTags("source")
+    @Get("/check-source")
```

Comment:
issue(blocking): now that I'm thinking, this is a DDOS vector and we should not continue. We can consider a few options like putting this behind RBAC for logged-in users.

But why aren't we just requesting from the client-side? Why we need to create a new endpoint for this?

Reply:
The endpoint was needed because of CORS restrictions, many sources block client-side requests. After, I can look with @lucaslobatob at some ways to make it more secure.

Reply:
I'm still not satisfied with the current solution, there are still too many vulnerabilities. Let's sync and create an epic documentation before proceeding.